### PR TITLE
Adding input value-channel for controlling spring-decompress cmd and output

### PR DESCRIPTION
**Pull request by asp8200**
Enable single fastq.gz-output from spring/decompress by input value-channel `write_one_fastq_gz`; no longer using `meta.single_end` for that.

Wanted over [here](https://github.com/nf-core/sarek/pull/1573).

## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Comment by SPPearce** [comment]
Does this make interleaved fastq files if it was paired end in the first place?

**Comment by SPPearce** [comment]
Any chance you could swap to nf-test while you are at it?

File: modules/nf-core/spring/decompress/meta.yml [hunk 1]
```diff
@@ -22,6 +22,10 @@ input:
       type: file
       description: Spring file to decompress.
       pattern: "*.{spring}"
+  - write_one_fastq_gz:
+      type: boolean
+      description: Controls whether spring should write one fastq.gz file with reads from both directionbs or two fastq.gz files with reads from distinct directions
```
Comment (maxulysse) [inline modules/nf-core/spring/decompress/meta.yml:27]:
```suggestion
      description: Controls whether spring should write one fastq.gz file with reads from both directions or two fastq.gz files with reads from distinct directions
```

**Review by maxulysse — DISMISSED** [review]

**Comment by SPPearce** [comment]
[The documentation says:](https://github.com/shubhamchandak94/Spring)
```
  -o [ --output-file ] arg        output file name (for paired end
                                  decompression, if only one file is specified,
                                  two output files will be created by suffixing
                                  .1 and .2.)
```
Does that mean that if the original file was paired end, it'll make two output files whatever?

**Comment by asp8200** [comment]
> [The documentation says:](https://github.com/shubhamchandak94/Spring)
> 
> ```
>   -o [ --output-file ] arg        output file name (for paired end
>                                   decompression, if only one file is specified,
>                                   two output files will be created by suffixing
>                                   .1 and .2.)
> ```
> 
> Does that mean that if the original file was paired end, it'll make two output files whatever?

I haven't tried that, but given the help text there, I would say 'yes' :-)  I could try it tomorrow. What I have tried is the followig: I ran
```
spring \
        -d \
        -g \
        -t 2 \
         \
        -i test_1.fastq.gz.spring \
        -o test_1_R1.fastq.gz test_1_R2.fastq.gz
```
where `test_1.fastq.gz.spring` only contained `R1`-reads. That resulted in `test_1_R1.fastq.gz` being written, while `test_1_R2.fastq.gz` wasn't even written.

Similar, I ran
```
spring \
        -d \
        -g \
        -t 2 \
         \
        -i test_2.fastq.gz.spring \
        -o test_2_R1.fastq.gz test_2_R2.fastq.gz
```
where `test_2.fastq.gz.spring` only contained R2-reads; again, `test_2_R1.fastq.gz` was written, while `test_2_R2.fastq.gz` wasn't.

**Comment by asp8200** [comment]
> [The documentation says:](https://github.com/shubhamchandak94/Spring)
> 
> ```
>   -o [ --output-file ] arg        output file name (for paired end
>                                   decompression, if only one file is specified,
>                                   two output files will be created by suffixing
>                                   .1 and .2.)
> ```
> 
> Does that mean that if the original file was paired end, it'll make two output files whatever?

So I got around to testing that, and the answer is 'yes': I ran spring on [this](https://github.com/nf-core/sarek/blob/ff5d37eb2f04af13a4d9f8b42abce2eee8d2e33c/tests/csv/3.0/bam_and_fastq_and_spring.csv?plain=1#L4) `.fastq.gz.spring`-file which contains both R1 and R2 reads:
```
spring -d -g -t 2 -i test_R1_R2.fastq.gz.spring -o foo
```
and that produced `foo.1` and `foo.2`.

**Comment by asp8200** [comment]
> Any chance you could swap to nf-test while you are at it?

@SPPearce @maxulysse : I added an nf-test. If your happy with that, then I'll get rid of the old test.

**Comment by SPPearce** [comment]
Ok, so do we need the value channel at all then?

**Comment by asp8200** [comment]
> Ok, so do we need the value channel at all then?

We might be able to get something working without the value channel, but then I guess the module would output fastg.gz-files with extension `.1` and `.2` on "interleaved" fastq.gz.spring input. Perhaps some downstream tools would complain about the fastq.gz.files not having extension different from the common `.fastq.gz`or `.fq.gz`? 🤔

**Review by maxulysse — APPROVED** [review]
Looks wonderful to me with the new tests. I approve

**Pull request merged**